dart/lib/src/scope.dart

```diff
@@ -60,7 +59,7 @@ class Scope {
   Map<String, dynamic> get contexts => Map.unmodifiable(_contexts);
 
   /// add an entry to the Scope's contexts
-  void setContexts(String key, dynamic value) {
+  void setContexts(String? key, dynamic value) {
```

Comment:
```suggestion
  void setContexts(String key, dynamic value) {
```

`if (key == null || value == null) return;` could be gone too